{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ea0b5780",
   "metadata": {},
   "source": [
    "# NGS Data Analysis and Visualization \n",
    "*Performing basic SAM file processing*\n",
    "\n",
    "Here I utilize the samtools package from bioconda and the IGV tools to read the FOXA2 gene.\n",
    "\n",
    "After creating the enviornemnt with:\n",
    "\n",
    "`conda create -n NGSAnalysis`\n",
    "\n",
    "`conda install bioconda::igv`\n",
    "\n",
    "`conda install bioconda::samtools`\n",
    "\n",
    "I `cd` to my data folder, create a new bash terminal with the activated environment, and run:\n",
    "\n",
    "`samtools view HG00096.chrom20.ILLUMINA.bwa.GBR.low_coverage.20120522.bam | awk {'if($4<22568101 && $4>22559642)print $0'} > FOXA2_reads.sam`\n",
    "\n",
    "Running the code reads the chromosome 20 file and extracts the FOXA2 reads into a seperate file called [FOXA2_reads.sam](data/processed/FOXA2_reads.sam), which found **304 reads**."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc0ede63",
   "metadata": {},
   "source": [
    "## Counting mate pairs\n",
    "\n",
    "Running:\n",
    "\n",
    "`sort -u -k1,1 FOXA2_reads.sam | wc` \n",
    "\n",
    "in the terminal gives the following numbers: **155 3328 69002**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d7983c24",
   "metadata": {},
   "source": [
    "## Plotting the distribution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b362a6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "df = pd.read_csv('data/processed/fragment_lengths.txt')\n",
    "\n",
    "x = np.arange(len(df))\n",
    "\n",
    "#plot\n",
    "fig, ax = plt.subplots(figsize=(10, 6))\n",
    "ax.bar(x, df['length'], color='skyblue', edgecolor='black')\n",
    "# Set x-ticks to the fragment lengths\n",
    "ax.set_xticks(x)\n",
    "ax.set_xticklabels(df['length'], rotation=90)\n",
    "ax.set_xlabel('Fragment Lengths')\n",
    "ax.set_ylabel('Count')\n",
    "ax.set_title('Fragment Length Distribution')\n",
    "plt.tight_layout()\n",
    "\n",
    "plt.show()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "plots",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.14.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
