{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ea0b5780",
   "metadata": {},
   "source": [
    "# NGS Data Analysis and Visualization \n",
    "*Performing basic SAM file processing*\n",
    "\n",
    "Here I utilize the samtools package from bioconda and the IGV tools to read the FOXA2 gene.\n",
    "\n",
    "After creating the enviornemnt with:\n",
    "\n",
    "`conda create -n NGSAnalysis`\n",
    "\n",
    "`conda install bioconda::igv`\n",
    "\n",
    "`conda install bioconda::samtools`\n",
    "\n",
    "I `cd` to my data folder, create a new bash terminal with the activated environment, and run:\n",
    "\n",
    "`samtools view HG00096.chrom20.ILLUMINA.bwa.GBR.low_coverage.20120522.bam | awk {'if($4<22568101 && $4>22559642)print $0'} > FOXA2_reads.sam`\n",
    "\n",
    "Running the code reads the chromosome 20 file and extracts the FOXA2 reads into a seperate file called [FOXA2_reads.sam](data/processed/FOXA2_reads.sam), which found **304 reads**."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc0ede63",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "variantvcf",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "name": "python",
   "version": "3.14.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
